src/mds/MDLog.cc

```diff
@@ -147,6 +147,16 @@ uint64_t MDLog::get_safe_pos() const
   return journaler->get_write_safe_pos(); 
 }
 
+void MDLog::_log_estimated_complete_time(utime_t start_time, uint64_t start_pos){
+  auto end_pos = journaler->get_write_pos();
+  auto current_pos = journaler->get_read_pos();
+  auto total_entries = end_pos - start_pos;
+  float percent_complete = (float) (current_pos - start_pos) / total_entries;
+  auto estimated_time = (1 - percent_complete) * (ceph_clock_now() - start_time);
+  dout(1) <<  setprecision(4) << "_replay status " << percent_complete * 100 << "\% complete" 
+    << " Estimated time remaining: " << estimated_time << " seconds" << dendl;
```

Comment:
A debug message is good but this ticket is really asking for a cluster health message. See `mds->clog->warn()` as used elsewhere.

I also think, in addition to those messages, a health warning should be sent via the MDS Beacon. That means you need to provide a method in `MDLog` for `Beacon` to call to get this same information. Create a unique health code for this situation.